packages/plans-grid-next/src/components/actions.tsx

```diff
@@ -97,6 +97,11 @@ const PlanFeatures2023GridActions = ( {
 		}
 	);
 
+	// availableForPurchase ultimately comes from use-check-plan-availability-for-purchase.ts
+	// which considers the plan of the current site. This is not appropriate for flows
+	// that are creating a new site, like /setup/new-hosted-site, so we override this value here.
+	availableForPurchase = availableForPurchase || ! currentSitePlanSlug;
```

Comment:
Is it possible to incorporate this logic in `use-check-plan-availability-for-purchase.ts`? If we can extend the conditions there to account for this, then it sounds better than overriding. The same hook is used for pricing.

Let me know if I should help more / if it proves unwieldy.

p.s. shouldn't `|| ! currentSitePlanSlug` be already part of the first condition (below)? 🤔 

```
// 1. No selected site or current plan: plan is in a context-free state and is available for purchase
if ( ! selectedSiteId || ! currentPlan ) {
    return [ planSlug, true ];
}

// 2. Otherwise, check if the plan is available for purchase on the selected site
return [ planSlug, isPlanAvailableForPurchase( state, selectedSiteId, planSlug ) ];
```

Reply:
I changed the approach, which fixes it for the hosting flows. I don't think it affects any others, so this change should be enough, but it would be good to double-check.